bcm53xx: switch to kernel 6.12

GoetzGoerisch opened the pull request:
switch the kernel to 6.12 by default

GoetzGoerisch committed:
bcm53xx: switch to kernel 6.12

switch the kernel to 6.12 by default

Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20874
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@GoetzGoerisch remember that you also have to remove 6.6 file... you can follow my commit for a similar ""template""

ricsc commented:
![R8500](https://github.com/user-attachments/assets/526a2ad3-7e47-46c9-8d54-82322b4fafc8)

I just compiled and flashed a build to a R8500. Looks good. Quick test confirms LAN/WAN/WIFI working.

GoetzGoerisch committed:
bcm53xx: drop support for kernel 6.6

Drop support for kernel 6.6 as now kernel 6.12 is set as default kernel
version.

Signed-off-by: Goetz Goerisch <ggoerisch@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20874
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Self-Hosting-Group commented:
Hello. Thank you for maintaining OpenWrt. For some time now, I have been unable to obtain an IP address for the WAN interface with snapshot builds (currently kernel 6.12.62), but the 24.10 releases work fine. `ifconfig` always displays `RX packets:0` for the WAN interface, but the DHCP server on the LAN works. Tested with an ASUS RT-AC68U router.

ricsc commented:
> Hello. Thank you for maintaining OpenWrt. For some time now, I have been unable to obtain an IP address for the WAN interface with snapshot builds (currently kernel 6.12.62), but the 24.10 releases work fine. `ifconfig` always displays `RX packets:0` for the WAN interface, but the DHCP server on the LAN works. Tested with an ASUS RT-AC68U router.

According to: #20167 RT-AC68U should work. You should open a thread in the community forum in order to troubleshoot.

GoetzGoerisch commented:
https://github.com/openwrt/openwrt/issues/21187 See issue

Self-Hosting-Group commented:
> #21187 See issue

The issue `RX packets:0` for the WAN interface on ASUS RT-AC68U still exists with snapshots and 25.12.0-rc4.
Is there anything we can do about it to get it fixed?

`ifconfig wan` shows the correct `HWaddr` but always `RX packets:0`/`RX bytes:0 (0.0 B)`

<details><summary>Show kernel logs</summary>

```
dmesg | grep b53-srab-switch
[    2.736146] b53-srab-switch 18007000.ethernet-switch: found switch: BCM53011, rev 2
[    2.984514] b53-srab-switch b53-srab-switch: error -EINVAL: invalid resource (null)
[    2.992228] b53-srab-switch b53-srab-switch: probe with driver b53-srab-switch failed with error -22
[    4.026272] b53-srab-switch 18007000.ethernet-switch: found switch: BCM53011, rev 2
[    4.098639] b53-srab-switch 18007000.ethernet-switch: OF node /ethernet-switch@18007000/ports/port@5 of CPU port 5 lacks the required "phy-mode" property
[    4.112407] b53-srab-switch 18007000.ethernet-switch: OF node /ethernet-switch@18007000/ports/port@5 of CPU port 5 lacks the required "phy-handle", "fixed-link" or "managed" properties
[    4.128838] b53-srab-switch 18007000.ethernet-switch: Skipping phylink registration for CPU port 5
[    4.139318] b53-srab-switch 18007000.ethernet-switch wan (uninitialized): PHY [dsa-0.0:00] driver [Generic PHY] (irq=POLL)
[    4.153021] b53-srab-switch 18007000.ethernet-switch lan1 (uninitialized): PHY [dsa-0.0:01] driver [Generic PHY] (irq=POLL)
[    4.166792] b53-srab-switch 18007000.ethernet-switch lan2 (uninitialized): PHY [dsa-0.0:02] driver [Generic PHY] (irq=POLL)
[    4.180541] b53-srab-switch 18007000.ethernet-switch lan3 (uninitialized): PHY [dsa-0.0:03] driver [Generic PHY] (irq=POLL)
[    4.194274] b53-srab-switch 18007000.ethernet-switch lan4 (uninitialized): PHY [dsa-0.0:04] driver [Generic PHY] (irq=POLL)
[    8.050801] b53-srab-switch 18007000.ethernet-switch lan1: configuring for phy/gmii link mode
[    8.070795] b53-srab-switch 18007000.ethernet-switch lan1: Link is Up - 1Gbps/Full - flow control rx/tx
[   12.887007] b53-srab-switch 18007000.ethernet-switch lan1: Link is Down
[   20.268871] b53-srab-switch 18007000.ethernet-switch lan1: configuring for phy/gmii link mode
[   20.316221] b53-srab-switch 18007000.ethernet-switch lan1: entered allmulticast mode
[   20.330543] b53-srab-switch 18007000.ethernet-switch lan1: entered promiscuous mode
[   20.388405] b53-srab-switch 18007000.ethernet-switch lan2: configuring for phy/gmii link mode
[   20.410996] b53-srab-switch 18007000.ethernet-switch lan2: entered allmulticast mode
[   20.419275] b53-srab-switch 18007000.ethernet-switch lan2: entered promiscuous mode
[   20.468080] b53-srab-switch 18007000.ethernet-switch lan3: configuring for phy/gmii link mode
[   20.497621] b53-srab-switch 18007000.ethernet-switch lan3: entered allmulticast mode
[   20.505941] b53-srab-switch 18007000.ethernet-switch lan3: entered promiscuous mode
[   20.536323] b53-srab-switch 18007000.ethernet-switch lan4: configuring for phy/gmii link mode
[   20.565019] b53-srab-switch 18007000.ethernet-switch lan4: entered allmulticast mode
[   20.574297] b53-srab-switch 18007000.ethernet-switch lan4: entered promiscuous mode
[   20.605981] b53-srab-switch 18007000.ethernet-switch wan: configuring for phy/gmii link mode
[   23.437747] b53-srab-switch 18007000.ethernet-switch lan1: Link is Up - 1Gbps/Full - flow control rx/tx
[   23.757768] b53-srab-switch 18007000.ethernet-switch wan: Link is Up - 1Gbps/Full - flow control off
```
</details>

GoetzGoerisch commented:
Please file an issue? No one will look in this merged PR.

GoetzGoerisch commented:
You should please file a complete new issue in the issue tracker not poluting a merged PR.
